app/helpers/form_errors_builder.rb

```diff
@@ -0,0 +1,29 @@
+class FormErrorsBuilder < ActionView::Helpers::FormBuilder
+  def error_messages
+    return unless object.respond_to?(:errors) && object.errors.any?
+
+    @template.content_tag :div, class: "panel panel-danger" do
+      error_header << error_panel
+    end
+  end
+
+private
+  def list_of_errors
```

Comment:
Prefer to define methods in order of calling - here `error_header`, `error_panel`, `list_of_errors`.